packages/uni_ui/lib/cards/restaurant_card.dart

```diff
@@ -0,0 +1,232 @@
+                    : UniIcon(PhosphorIconsRegular.heart,
+                        color: Theme.of(context).primaryColor)),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class RestaurantCardMenuItem extends StatelessWidget {
+  const RestaurantCardMenuItem(
+      {super.key, required this.menuItemName, required this.menuItemType});
+
+  final String menuItemName;
+  final String menuItemType;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 0, vertical: 8.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          Expanded(
+              flex: 1,
+              child: MenuItemIcons.getMenuItemIcon(context, menuItemType)),
+          Expanded(
+            flex: 5,
+            child: Text(
+              menuItemName,
+              style: Theme.of(context).textTheme.bodyLarge,
+              softWrap: true,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class RestaurantIcons {
+  static const cantina = PhosphorIconsDuotone.cookingPot;
+  static const grill = PhosphorIconsDuotone.cookingPot;
+  static const restaurant = PhosphorIconsDuotone.chefHat;
+  static const snackBar = PhosphorIconsDuotone.hamburger;
+
+  static UniIcon getRestaurantIcon(
+      BuildContext context, String restaurantType) {
+    var iconType;
+    switch (restaurantType) {
+      case "cantina":
+        iconType = cantina;
+        break;
+      case "grill":
+        iconType = grill;
+        break;
+      case "restaurant":
+        iconType = restaurant;
+        break;
+      case "snackBar":
+        iconType = snackBar;
+        break;
+      default:
+        iconType = restaurant;
+    }
+    return UniIcon(iconType, color: Theme.of(context).primaryColor);
+  }
+}
+
+class MenuItemIcons {
+  static const cow = PhosphorIconsDuotone.cow;
+  static const fish = PhosphorIconsDuotone.fish;
+  static const vegetarian = PhosphorIconsDuotone.plant;
+  static const soup = PhosphorIconsDuotone.bowlSteam;
+
+  static UniIcon getMenuItemIcon(BuildContext context, String menuItemType) {
+    var iconType;
+    switch (menuItemType) {
+      case "cow":
+        iconType = cow;
+        break;
+      case "fish":
+        iconType = fish;
+        break;
+      case "vegetarian":
+        iconType = vegetarian;
+        break;
+      case "soup":
+        iconType = soup;
+        break;
+      default:
+        iconType = cow;
+    }
+    return UniIcon(iconType, color: Theme.of(context).primaryColor);
+  }
+}
+
+class UniIcon extends PhosphorIcon {
+  const UniIcon(
+    IconData icon, {
+    super.key,
+    double size = 24,
+    Color? color,
+    String? semanticLabel,
+    TextDirection? textDirection,
+  }) : super(
+          icon,
+          size: size,
+          color: color,
+          semanticLabel: semanticLabel,
+          textDirection: textDirection,
+        );
+}
+
+class MenuItem {
```

Comment:
To avoid introducing models in the `uni_ui` package, I suggest creating a MenuItem widget in uni_ui instead of defining a MenuItem model (that would take parameters like `icon` and `name`) and handle the display logic for each menu item. The `RestaurantCard` would then receive a list of `MenuItem` widgets, and uni_app would be responsible for creating and passing this list.
What do you think?

Reply:
I'm a bit skeptical and I do not like the idea of having models in the ui, so you got a point there. However, if I understood what you're saying, we'd have to create a model in the ui and follow the following path: **create in ui -> instantiate in app -> pass to ui**, which is a bit strange.
Either way, or that or passing all the information destructured.

Reply:
To clarify my point: instead of defining any model in the `uni_ui` package, we should create a widget that takes attributes like `icon` and `name` as parameters. The `RestaurantCard` would then receive a list of these widgets.

Reply:
I agree with @vitormpp. MenuItem can be a widget and the restaurant card receives a list of MenuItem widgets